app/src/main/java/org/openobservatory/ooniprobe/activity/RunningActivity.java

```diff
@@ -174,19 +171,20 @@ protected void onResume() {
             return;
         }
         IntentFilter filter = new IntentFilter("org.openobservatory.ooniprobe.activity.RunningActivity");
-        receiver = new TestRunBroadRequestReceiver();
+        receiver = new TestRunBroadRequestReceiver(preferenceManager, new TestRunnerEventListener());
         LocalBroadcastManager.getInstance(this).registerReceiver(receiver, filter);
         //Bind the RunTestService
         Intent intent = new Intent(this, RunTestService.class);
-        bindService(intent, this, Context.BIND_AUTO_CREATE);
+//        bindService(intent, this, Context.BIND_AUTO_CREATE);
+        bindService(intent, receiver, Context.BIND_AUTO_CREATE);
     }
 
     @Override
     protected void onPause() {
         super.onPause();
-        if (isBound) {
-            unbindService(this);
-            isBound = false;
+        if (receiver.isBound()) {
+            unbindService(receiver);
+            receiver.setBound(false);
```

Comment:
this would mean passing a context object to the `receiver`  so it can be able to call `unbindService(connection)` once the checks are performed.
That did not strike me as the way to handle the situation. passing an additional argument to the class just to be used in one case for convenience.

Reply:
✔️